File: Minimum-Depth-of-Binary-Tree/Minimum-Depth-of-Binary-Tree.cpp

/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode *root) {
        if(!root)
            return 0;
        else if(!root->left && !root->right)
            return 1;
        else if(root->left && root->right)
            return min(minDepth(root->left),minDepth(root->right))+1;
        else if(root->left)
            return minDepth(root->left)+1;
        else
            return minDepth(root->right)+1;
    }
};


 

File: solutions/Minimum-Depth-of-Binary-Tree/Minimum-Depth-of-Binary-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function (root) {
  if (root === null) {
    return 0;
  }
  const q = [root];
  let ans = 0;
  while (q.length > 0) {
    ans++;
    const length = q.length;
    for (let i = 0; i < length; i++) {
      const temp = q.shift();
      if (temp.left !== null) {
        q.push(temp.left);
      }
      if (temp.right !== null) {
        q.push(temp.right);
      }
      if (temp.left === null && temp.right === null) {
        return ans;
      }
      q.unshift();
    }
  }
};